lib/commands/doctor/checks/system-stack.js

```diff
@@ -7,21 +7,10 @@ const taskTitle = 'Checking system compatibility';
 const nginxProgramName = process.env.NGINX_PROGRAM_NAME || 'nginx';
 const versionRegex = /^(?:16|18|20)/;
 
-async function hasSystemd() {
+async function hasService(name) {
     try {
-        const {list} = await sysinfo.processes();
-        const rootProcess = list.find(proc => proc.pid === 1);
-
-        return Boolean(rootProcess) && rootProcess.name === 'systemd';
-    } catch (error) {
-        return false;
-    }
-}
-
-async function hasNginx() {
-    try {
-        const services = await sysinfo.services('*');
-        return services.some(s => s.name === nginxProgramName);
+        const services = await sysinfo.services(name);
```

Comment:
Not sure how long this takes; do we want to cache it?

Reply:
nah, it's a fairly straightforward `ps` call that does some grepping on the output, so it shouldn't take long at all to run.